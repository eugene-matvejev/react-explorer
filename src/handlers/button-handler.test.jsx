import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ButtonHandler from './button-handler';

configure({ adapter: new Adapter() });

describe('<ButtonHandler/>', () => {
    const props = {
        data: [
            {
                name: 'Add',
            },
        ],
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<ButtonHandler {...props} />);

            expect(c).toMatchSnapshot();
        });

        describe('with optional props', () => {
            [
                ['data-cy', '{{data-cy}}'], /** should be passed 'as is' */
            ].forEach(([prop, v]) => {
                it(`[::${prop}] as "${v}"`, () => {
                    const c = shallow(<ButtonHandler {...props} {...{ [prop]: v }} />);

                    expect(c).toMatchSnapshot();
                });
            });
        });
    });
});

