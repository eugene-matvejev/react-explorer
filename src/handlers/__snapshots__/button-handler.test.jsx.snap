// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonHandler/> render with default/required props 1`] = `
<section
  className="button-handler"
>
  <Button
    className=""
    data-cy="tile-0"
    key="0"
    label=""
    name="ADD"
  />
  <Button
    className=""
    data-cy="tile-1"
    key="1"
    label=""
    name="FILTER"
  />
</section>
`;

exports[`<ButtonHandler/> render with optional props [::data-cy] as "{{data-cy}}" 1`] = `
<section
  className="button-handler"
>
  <Button
    className=""
    data-cy="tile-0"
    key="0"
    label=""
    name="ADD"
  />
  <Button
    className=""
    data-cy="tile-1"
    key="1"
    label=""
    name="FILTER"
  />
</section>
`;
